shiny/express/_is_express.py

```diff
@@ -78,3 +88,24 @@ def visit_Module(self, node: ast.Module) -> None:
     # Don't recurse into any nodes, so the we'll only ever look at top-level nodes.
     def generic_visit(self, node: ast.AST) -> None:
         pass
+
+
+def find_magic_comment_mode(content: str) -> Literal["core", "express"] | None:
+    """
+    Look for a magic comment of the form "# shiny_mode: express" or "# shiny_mode:
+    core".
+
+    Returns
+    -------
+    :
+        `True` if Shiny Express comment is found, `False` if Shiny Core comment is
+        found, and `None` if no magic comment is found.
+    """
+
+    for line in content[:1000].splitlines():
```

Comment:
If you use the `re.MULTILINE` flag you don't need to split (although you'd need to change `\s` to `[ \t]`